### ISSUE TITLE
Zig fails to build lib on macOS

### ISSUE THREAD

**Issue by euantorano** [ziglang/zig#1878]
Creating a new library with `init-lib` and then attempting to build it on macOS Mojave 10.14.2 fails.

Steps:

```
mkdir zig-test && cd zig-test
zig init-lib
zig build
```

Output:

```
Undefined symbol: /Users/euantorano/Desktop/zig-crash-test/zig-cache/zig-crash-test.o: __mh_execute_header
symbol(s) not found
lld: warning: ignoring unknown argument: -headerpad_max_install_names
lld: warning: -sdk_version is required when emitting min version load command.  Setting sdk version to match provided min version

The following command exited with error code 1:
zig build-lib /Users/euantorano/Desktop/zig-crash-test/src/main.zig --cache-dir /Users/euantorano/Desktop/zig-crash-test/zig-cache --output /Users/euantorano/Desktop/zig-crash-test/zig-cache/libzig-crash-test.a --output-h /Users/euantorano/Desktop/zig-crash-test/zig-cache/zig-crash-test.h --name zig-crash-test

Build failed. The following command failed:
/Users/euantorano/Library/Application Support/zig/stage1/artifact/_rGLoCx9ObHWbibERqfIOmBmsy5bDjGpDd4abDyi3PGZpQIQRdgroELQ5n1mBwRO/build zig /Users/euantorano/Desktop/zig-crash-test /Users/euantorano/Desktop/zig-crash-test/zig-cache
```

Contents of the workspace after the build:

```
.
├── build.zig
├── src
│   └── main.zig
└── zig-cache
    ├── zig-crash-test.h
    └── zig-crash-test.o

2 directories, 4 files
```

**Comment by andrewrk** [comment]
Hmm this is interesting. I thought that symbol was guaranteed in libSystem. I'll have to investigate.

**Comment by wilsonk** [comment]
@andrewrk I think this was fixed with 28bf768 wasn't it?

**Comment by andrewrk** [comment]
That's right. See also #1981 and #1982
